ddtrace/span.py

```diff
@@ -416,7 +416,18 @@ def set_metric(self, key, value):
 
         if key in self._meta:
             del self._meta[key]
-        self.metrics[key] = value
+        self._metrics[key] = value
+
+    @removed_property(
+        message="Use Span.get_metric or Span.set_metric instead.",
+        removal_version="1.0.0",
+    )
+    def metrics(self):
+        return self._metrics
+
+    @metrics.setter  # type: ignore[no-redef]
+    def metrics(self, value):
+        self._metrics = value
```

Comment:
Do we have any tests for these?

It seems there was a small bug `self._metric`/`self._metrics`, but this wasn't caught in any tests?

Reply:
Nope. All the tests were updated so they did not depend on the deprecated logic. To catch these errors we've been mainly relying on code reviews. Should we start adding tests for deprecated methods/attributes?

Reply:
yes, we need tests that the deprecated properties work as expected, otherwise we are likely to introduce bugs into our customers code who still rely on the deprecated APIs